tools/cli/commands/delete.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Methods for implementing the `datalab delete` command."""
+
+import argparse
+
+
+description=(
+"""{0} {1} deletes the given Datalab instance's
+Google Compute Engine VM.
+
+By default, the persistent disk's auto-delete configuration determines
+whether or not that disk is also deleted.
+
+If you wish to override that setting, you can pass in one of either the
+`--delete-disk` flag or the `--keep-disk` flag.
+
+For more information on disk auto-deletion, see
+https://cloud.google.com/compute/docs/disks/persistent-disks#updateautodelete
+"""
+)
+
+
+_DELETE_DISK_HELP=(
+"""Whether or not to delete the instance's persistent disk
+regardless of the disks' auto-delete configuration."""
+)
+
+
+_KEEP_DISK_HELP=(
+"""Whether or not to keep the instance's persistent disk
+regardless of the disks' auto-delete configuration."""
+)
+
+
+def flags(parser):
+    """Add command line flags for the `delete` subcommand.
+
+    Args:
+      parser: The argparse parser to which to add the flags.
+    """
+    parser.add_argument(
+        'instance',
+        metavar='INSTANCE',
+        help='name of the instance to delete')
+
+    auto_delete_override = parser.add_mutually_exclusive_group()
```

Comment:
This one is a bit different from the others, as a user might have to override to get back the default behavior.

Specifically, they could create an instance, and then (via the console) change the attached disk to be deleted. Alternatively, we may make the disk keep/delete config a flag in the create command.

In those scenarios, if they want the `datalab delete` command to not delete the disk, they need a flag to get back to the default behavior.